### PR TITLE
Added search and reformated the workflow pending requests UI for admin dashboard

### PR DESCRIPTION
**Pull request by SahanHerath**

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/APIStateChange.jsx [hunk 1]
```diff
@@ -97,10 +91,22 @@ function ListLabels() {
...
+                        apiProvider: obj.properties.currentState,
+                        referenceId: obj.referenceId,
+                        createdTime: obj.createdTime,
+                        properties: obj.properties,
+                    };
+                });
+                return workflowlist;
             })
             .catch((error) => {
+                Alert.error('Unable to get workflow pending requests for API State Change', error.message);
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/APIStateChange.jsx:109]: Shall we Internationalise the static message?

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/APIStateChange.jsx [hunk 2]
```diff
@@ -113,8 +119,8 @@ function ListLabels() {
             setData(LocalData);
         })
             .catch((e) => {
-                Alert.error('Unable to fetch data ' + e);
-                console.error('Unable to fetch data ' + e);
+                Alert.error('Unable to fetch data. ', e.message);
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/APIStateChange.jsx:122]: Internationalize the static message

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/APIStateChange.jsx [hunk 3]
```diff
@@ -123,10 +129,13 @@ function ListLabels() {
     }, []);
 
     const updateStatus = (referenceId, value) => {
-        const body = {};
-        body.status = value;
-        body.attributes = {};
-        body.description = 'Approve workflow request.';
+        const body = { status: value, attributes: {}, description: '' };
+        if (value === 'APPROVED') {
+            body.description = 'Approve workflow request.';
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/APIStateChange.jsx:134]: Internationalise these texts

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/APIStateChange.jsx [hunk 4]
```diff
@@ -141,7 +150,7 @@ function ListLabels() {
             .catch((error) => {
                 const { response } = error;
                 if (response.body) {
-                    response.body.description = 'Unable to complete API state change approve/reject process';
+                    Alert.error('Unable to complete API state change approve/reject process. ', error.message);
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/APIStateChange.jsx:153]: Internationalise static message

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/ApplicationCreation.jsx [hunk 5]
```diff
@@ -102,10 +95,21 @@ function ListLabels() {
...
+                        userName: obj.properties.userName,
+                        referenceId: obj.referenceId,
+                        createdTime: obj.createdTime,
+                        properties: obj.properties,
+                    };
+                });
+                return workflowlist;
             })
             .catch((error) => {
+                Alert.error('Unable to get workflow pending requests for Application Creation', error.message);
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/ApplicationCreation.jsx:112]: Internationalise static message

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/ApplicationCreation.jsx [hunk 6]
```diff
@@ -118,8 +122,8 @@ function ListLabels() {
             setData(LocalData);
         })
             .catch((e) => {
-                Alert.error('Unable to fetch data ' + e);
-                console.error('Unable to fetch data ' + e);
+                Alert.error('Unable to fetch data. ', e.message);
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/ApplicationCreation.jsx:125]: Internationalise static message

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/APIStateChange.jsx (hunk 2 above)
Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/APIStateChange.jsx:122]: Why are we showing a static error and the error from backend as well?

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/ApplicationCreation.jsx (hunk 5 above)
Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/ApplicationCreation.jsx:112]: Why are we showing a static error and error from backend as well? Check these in other places as well

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/ApplicationCreation.jsx [hunk 7]
```diff
@@ -146,7 +153,7 @@ function ListLabels() {
             .catch((error) => {
                 const { response } = error;
                 if (response.body) {
-                    response.body.description = 'Unable to complete application creation approve/reject process';
+                    Alert.error('Unable to complete application creation approve/reject process. ', error.message);
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/ApplicationCreation.jsx:156]: same here

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/ApplicationCreation.jsx [hunk 8]
```diff
@@ -228,24 +227,29 @@ function ListLabels() {
...
-            name: 'creater',
+            name: 'userName',
             label: intl.formatMessage({
-                id: 'Workflow.ApplicationCreation.table.header.Creater',
+                id: 'Workflow.ApplicationCreation.table.header.userName',
                 defaultMessage: 'Created by',
             }),
             options: {
+                sort: false,
+                /* customSort: (data1, colIndex, order) => {
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/ApplicationCreation.jsx:241]: Shall we remove commented code?

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/ApplicationCreation.jsx [hunk 9]
```diff
@@ -332,6 +334,7 @@ function ListLabels() {
 
     const columns = [
         ...columProps,
+
```
Comment (kavishkafernando) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/ApplicationCreation.jsx:337]: Remove extra line

Comment (SahanHerath) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/ApplicationCreation.jsx]: fixed with 9c884f8

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/ApplicationCreation.jsx (hunk 8 above)
Comment (SahanHerath) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/ApplicationCreation.jsx]: fixed with 9c884f8

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/ApplicationCreation.jsx (hunk 7 above)
Comment (SahanHerath) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/ApplicationCreation.jsx]: fixed with 9c884f8

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/ApplicationCreation.jsx (hunk 6 above)
Comment (SahanHerath) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/ApplicationCreation.jsx]: fixed with 9c884f8

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/ApplicationCreation.jsx (hunk 5 above)
Comment (SahanHerath) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/ApplicationCreation.jsx]: fixed with 9c884f8. error from the backend is removed from alert.

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/APIStateChange.jsx (hunk 4 above)
Comment (SahanHerath) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/APIStateChange.jsx]: fixed with 9c884f8

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/APIStateChange.jsx (hunk 3 above)
Comment (SahanHerath) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/APIStateChange.jsx:140]: the description is a value sent to the database. So didn't update as a static message.

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/APIStateChange.jsx (hunk 2 above)
Comment (SahanHerath) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/APIStateChange.jsx]: fixed with 9c884f8

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/APIStateChange.jsx (hunk 1 above)
Comment (SahanHerath) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Workflow/APIStateChange.jsx]: fixed with 9c884f8

**Pull request merged**